code/dynamic_programming/src/trapping_rain_water/trapping_rain_water.java

```diff
@@ -0,0 +1,36 @@
+public class Main {
+    public static void main(String args[]) {
+        int[] heightArray =  {1, 2, 1, 3, 1, 2, 1, 4, 1, 0, 0, 2, 1, 4};
+        System.out.println(trap_rain_water(heightArray));
+    }
+
+    public static int trap_rain_water(int[] height) {
+        int result = 0;
+    
+        if(height==null)
+            return result;
+    
+        int left[] = new int[height.length];
+        int right[]= new int[height.length];
```

Comment:
Please be consistent in formatting code.

Reply:
I have updated the changes suggested by you. Can you review it again.